### PR TITLE
Bundle or remove external libs

### PR DESCRIPTION
**Pull request by Mohammer5**
# Changes overview
I've removed the external scripts as they were either bundled anyways (react, rxjs) or not used whatsoever (lodash, react addons).

# Complications
The bundled react version was (and still is) 16, but due to the fact that react v15 was provided as external script and some components were written in a deprecated way, I had to transform them to ES6 components.

# Work arounds
Unfortunately (or luckily ;)) react v16 components don't support mixins anymore, so I had to create Higher Order Components inside this project although the translation mixin was provided by the d2-ui library. They are inside `.HOC.js` files.

### CODE REVIEW

File: src/Translate.HOC.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default component => {
```
Comment (Birkbjo) [inline src/Translate.HOC.js]: Could you add a displayname to the component for easier debugging?

See https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging

File: src/form-fields/check-box.js [hunk 2]
```diff
@@ -1,17 +1,13 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import Checkbox from 'material-ui/Checkbox';
-
+import applyMuiThemeContext from '../mui-theme.HOC.js'
 import MuiThemeMixin from '../mui-theme.mixin';
```
Comment (Birkbjo) [inline src/form-fields/check-box.js]: Can remove this now, right?

File: src/mui-theme.mixin.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import React from 'react';
```
Comment (Birkbjo) [inline src/mui-theme.mixin.js]: Should be able to remove this file.

File: src/mui-theme.HOC.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AppTheme from './theme';
+
+export default component => {
```
Comment (Birkbjo) [inline src/mui-theme.HOC.js]: I'm not too familiar with how this app has configured material ui. What's the difference between this and the MuiThemeProvider provided by material-ui/styles/MuiThemeProvider? It's almost the same as the original source: https://github.com/mui-org/material-ui/blob/v0.x/src/styles/MuiThemeProvider.js

I find it strange that multiple components need to add this to the context. Should be enough with a provider at the root-level of the app.

File: src/Translate.HOC.js (hunk 1 above)
Comment (Mohammer5) [reply src/Translate.HOC.js]: Ah, that's a good-read, thanks for that, will add the display name.

File: src/mui-theme.mixin.js (hunk 3 above)
Comment (Mohammer5) [reply src/mui-theme.mixin.js]: Yes..

File: src/mui-theme.HOC.js (hunk 4 above)
Comment (Mohammer5) [reply src/mui-theme.HOC.js]: That's a good questions.
I have to look into that and see if we can simply use the Provider on a higher leven and then get rid of the MuiTheme HOC completely, would be awesome

File: src/app.component.js [hunk 5]
```diff
@@ -154,7 +155,7 @@ class AppComponent extends React.Component {
                         currentSection={this.state.category}
                         showSearchField
                         searchFieldLabel={this.props.d2.i18n.getTranslation('search_settings')}
-                        ref={setSidebar}
+                        ref={setSidebar} // eslint-disable-line react/jsx-no-bind
```
Comment (Birkbjo) [inline src/app.component.js]: I would just move setSidebar out to a method, so we don't need this. Also cleaner then having a function declared in the render-method.

**Pull request merged**